htdocs/core/js/lib_foot.js.php

```diff
@@ -101,320 +98,313 @@
-							  //$(".dropdown dt a").parents(\'form:first\').submit();
-							  //$(".dropdown dt a").closest("form").submit();
-							  lastopendropdown.closest("form").submit();
-					      }
-					  }
-
-                      // Note: Did not find a way to get exact height (value is update at exit) so i calculate a generic from nb of lines
-                      heigthofcontent = 21 * $(this).parent().parent().find(\'dd div ul li\').length;
-                      if (heigthofcontent > 300) heigthofcontent = 300; // limited by max-height on css .dropdown dd ul
-                      posbottom = $(this).parent().parent().find(\'dd\').offset().top + heigthofcontent + 8;
-                      var scrollBottom = $(window).scrollTop() + $(window).height();
-                      diffoutsidebottom = (posbottom - scrollBottom);
-                      console.log("heigthofcontent="+heigthofcontent+", diffoutsidebottom (posbottom="+posbottom+" - scrollBottom="+scrollBottom+") = "+diffoutsidebottom);
-                      if (diffoutsidebottom > 0)
-                      {
-                            pix = "-"+(diffoutsidebottom+8)+"px";
-                            console.log("We reposition top by "+pix);
-                            $(this).parent().parent().find(\'dd\').css("top", pix);
-                      }
-                  });
-
-                  // Click on a link into the popup "link to" or other dropdown that ask to close drop down on element click, so close dropdown
-                  $(".dropdowncloseonclick").on(\'click\', function () {
-                      console.log("Link has class dropdowncloseonclick, so we close/hide the popup ul");
-                      //$(this).parent().parent().hide();		// $(this).parent().parent() is ul
-					  $(this).parent().parent().removeClass("open");	// $(this).parent().parent() is ul
-                  });
-
-				  // Click outside of any dropdown
-                  $(document).bind(\'click\', function (e) {
-                      var $clicked = $(e.target);	// This is element we click on
-                      if (!$clicked.parents().hasClass("dropdown")) {
-                          //console.log("close dropdown dd ul - we click outside");
-						  //$(".dropdown dd ul").hide();
-						  $(".dropdown dd ul").removeClass("open");
-
-						  if ($("input:hidden[name=formfilteraction]").val() == "listafterchangingselectedfields") {
-							  console.log("resubmit form saved into lastopendropdown after clicking outside of dropdown and having change selectlist from selectlist field of hamburger dropdown");
-							  //$(".dropdown dt a").parents(\'form:first\').submit();
-							  //$(".dropdown dt a").closest("form").submit();
-							  lastopendropdown.closest("form").submit();
-					      }
-					  }
-                  });
-              });
-           ';
-}
+<?php if (!defined('JS_JQUERY_DISABLE_DROPDOWN')) : ?>
```

Comment:
The use of 
```
if ... :
endif
```
 is not allowed by the php syntax rules, must follow the PSR-12 rules (section 5.1) so the { and }

Reply:
This notation is used when inserting PHP code into HTML / JS files, so I'm not sure that this PSR-12 recommendation applies in this case. Using classic notation in js / html files makes them less readable, whereas we're looking for a shortened “template engine” type of writing in this case.

Reply:
Also, the notation <?= ?> is apparently forbidden in the Dolibarr code, but this notation is PSR-12 friendly, and is always available (independant from short_open_tag : https://www.php.net/manual/en/ini.core.php#ini.short-open-tag). It should be authorized, so we can work on real html/php template-like files